rpc/types/src/macros.rs

```diff
@@ -59,177 +57,234 @@ macro_rules! define_request_and_response {
         $monero_code_line_end:literal,
 
         // The base `struct` name.
+        // Attributes added here will apply to _both_
+        // request and response types.
+        $( #[$type_attr:meta] )*
         $type_name:ident,
 
+        // The request type (and any doc comments, derives, etc).
+        $( #[$request_type_attr:meta] )*
+        Request {
+            // And any fields.
+            $(
+                $( #[$request_field_attr:meta] )*
+                $request_field:ident: $request_field_type:ty $(= $request_field_type_default:expr)?,
```

Comment:
And also, it supports `cuprate_epee_encoding`'s default value syntax.

We can find a way to automatically do this with `serde` too (right now the caller must add the serde attribute) but that's for another PR.